nucypher/config/characters.py

```diff
@@ -76,14 +78,27 @@ def __init__(self,
         self.availability_check = availability_check if availability_check is not None else self.DEFAULT_AVAILABILITY_CHECKS
         super().__init__(dev_mode=dev_mode, *args, **kwargs)
 
+    @classmethod
+    def checksum_address_from_filepath(cls, filepath: str) -> str:
+        # must always extract worker address by "peeking" inside the ursula configuration file.
+        checksum_address = cls.peek(filepath=filepath, field='checksum_address')
+        federated = bool(cls.peek(filepath=filepath, field='federated_only'))
+        if not federated:
+            checksum_address = cls.peek(filepath=filepath, field='worker_address')
+
+        if not is_checksum_address(checksum_address):
+            raise RuntimeError(
+                f"Invalid checksum address detected in configuration file at '{filepath}'.")
```

Comment:
Odd looking newline here.

```suggestion
            raise RuntimeError(f"Invalid checksum address detected in configuration file at '{filepath}'.")
```